zebrad/src/components/sync.rs

```diff
@@ -95,7 +76,9 @@ const TIPS_RETRY_TIMEOUT: Duration = Duration::from_secs(60);
 ///
 /// This timeout is particularly important on instances with slow or unreliable
 /// networks, and on testnet, which has a small number of slow peers.
-const SYNC_RESTART_TIMEOUT: Duration = Duration::from_secs(100);
+const SYNC_RESTART_TIMEOUT: Duration = Duration::from_secs(45);
```

Comment:
@teor2345 I turned this down to accelerate the restart process.  It seems to still work on testnet.  Although it gets stuck sometimes, it does eventually make progress.  We can turn it back up if problems re-occur, but it seemed like faster restarts on mainnet were more important.

Reply:
I'd really like to reduce the sync timeout, but I'd also like to make sure it works across a range of network conditions.

Have you tested it from a low bandwidth or high latency network location?

If not, I can do those tests tomorrow.

Reply:
In particular, I'm concerned about network-limited nodes and download floods.

But hopefully DownloadSet reduces duplicates, and Hedge reduces failures,

Reply:
Our plan is to:
- [ ] Test this change on a slower network using mainnet
- [ ] Test this change on testnet, but treat stalls as a lower priority, because testnet doesn't have many nodes
- [ ] Deploy some extra `zcashd` nodes on testnet, then re-test this change on testnet

Reply:
![zebrad mainnet stall](https://user-images.githubusercontent.com/8951843/96960963-fce9b200-1546-11eb-9de9-515844bfbe90.png)

I'm seeing frequent unrecoverable stalls on mainnet.

I suspect that the sync parameters are over-tuned for high-bandwidth, low-latency, reliable networks.

Reply:
This might actually be an issue with the block verifier or state service, I'll retry with an empty state.
![Block Verifier](https://user-images.githubusercontent.com/8951843/96964601-1b9f7700-154e-11eb-8a7b-a989c3dc69c3.png)

Reply:
Yep, 420.3K is the sapling checkpoint height, so it's a state or block verifier issue.

Reply:
Let's merge this, so the sync works, and use it to fix the verifier pipelines.